medallion/backends/memory_backend.py

```diff
@@ -213,6 +215,38 @@ def get_object(self, api_root, collection_id, object_id, filter_args, allowed_fi
             )
             return create_resource("objects", objs), {}
 
+    def delete_object(self, api_root, collection_id, obj_id, filter_args, allowed_filters):
+        if api_root in self.data:
+            api_info = self._get(api_root)
+            collections = api_info.get("collections", [])
+            objs = []
+            manifests = []
+            for collection in collections:
+                if "id" in collection and collection_id == collection["id"]:
+                    coll = collection.get("objects", [])
+                    for obj in coll:
+                        if obj_id == obj["id"]:
+                            objs.append(obj)
+                    manifests = collection.get("manifest", [])
+                    break
+            if len(objs) == 0:
```

Comment:
This check should be placed after the `full_filter.process_filter()` call and before the for-loop. Depending on the filters provided `objs` may be an empty list, perform no action and still return a 200 Response.

Reply:
If no objects are returned in the for loop, that means no objects with that ID exist in the collection. Filters don't get checked until after this. The spec says that if no objects with that ID exist, then a 404 is supposed to be returned.

Reply:
Right, we are not removing the check just moving it down. Otherwise you will need to do the same check twice since. The logical error is that after the filters being processed it can still return a 200 OK for an object that was not really deleted.

Reply:
Ah I see now. So if after the filters are ran, there are no objects to delete, it should throw a 404? I assumed it would have the same behavior as getting an object, where if a filter ends up removing all objects, just an empty envelope is returned instead of a 404.

Reply:
Yes, even after the filters ran it should throw a 404 if no object(s) matched the request. 

On the other hand, keep in mind that you could make a request that deletes all versions of an object given the hypothetical request: DELETE /api1/collections/91a7b528-80eb-42ed-a74d-c6fbd5a26116/object/
indicator--252c7c11-daf2-42bd-843b-be65edca9f61/?match[version]=all HTTP/1.1 and the response should be 200 OK if it deleted them successfully. There is no payload on successful requests to this Endpoint